clients/setup/setup.go

```diff
@@ -151,6 +152,25 @@ func (c Client) onboardingRequest(params *Params) (req api.OnboardingRequest, er
 	req.RetentionPeriodSeconds = &rpSecs
 
 	if params.Force {
+		// The server doesn't give a very helpful error message when required fields are missing,
```

Comment:
Would it be practical at all to update the server code so that it gives better error messages in response rather than duplicating the check?

Reply:
Looks like it'd be easy enough to apply the same fix on the server side. Interestingly, it looks like the password isn't actually required by the API even though the CLI has historically forced one to be set...

Reply:
I'll back this out for now so we can merge the 1st commit, which is needed for `influxd upgrade` to work in PowerShell